runtime/arc.js

```diff
@@ -102,11 +102,12 @@ class Arc {
     return [...this._viewsById.values()];
   }
 
-  clone() {
+  // Makes a copy of the arc used for speculative execution.
+  makeSpeculative() {
     var arc = new Arc({loader: this._loader, id: this.generateID(), pecFactory: this._pecFactory});
     var viewMap = new Map();
     this._views.forEach(v => viewMap.set(v, v.clone()));
-    arc.particles = this.particles.map(p => p.clone(viewMap));
+    // TODO: do we need to clone particles?
```

Comment:
Yes, we absolutely do :)

This is a data structure that tracks particle connections, we don't want any changes in the clone to leak back into the original arc.